bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -1227,8 +1260,24 @@ public Rectangle getBounds() {
 	return getBounds (100);
 }
 
-Rectangle getBounds(int zoom) {
-	Rectangle bounds = getBoundsInPixels();
+/**
+ * Returns the bounds of the receiver scaled to the specified zoom level.
+ * The rectangle will always have x and y values of 0, and the width and
+ * height of the image scaled to the given zoom level.
+ *
+ * @param zoom zoom level the image bounds shall be scaled for
+ *
+ * @return a rectangle specifying the image's bounds in pixel for the given zoom level.
+ *
+ * @exception SWTException <ul>
+ *    <li>ERROR_GRAPHIC_DISPOSED - if the receiver has been disposed</li>
+ *    <li>ERROR_INVALID_IMAGE - if the image is not a bitmap or an icon</li>
+ * </ul>
+ * @noreference This method is not intended to be referenced by clients.
+ */
+public Rectangle getBounds(int zoom) {
```

Comment:
By using `getBoundsInPixelsFromNative()` instead of `getBoundsInPixels()`, there is no dispose check anymore, so the defined contract for this method to throw an exception in case the receiver is disposed will probably not hold (and maybe even lead to unexpected behavior when trying to calculate the bounds for a disposed image).

Reply:
So adding a dispose check there should be sufficient?

Reply:
Yes, I think so.